src/os_crypto/aes/aes_op.c

```diff
@@ -95,7 +95,7 @@ int decrypt_AES(const unsigned char *ciphertext, int ciphertext_len, unsigned ch
     }
 
 	plaintext_len += len;
-
+    plaintext[plaintext_len] = '\0';
```

Comment:
This is already managed in `ReadSecMSG()` (` cleartext[buffer_size] = '\0';`).

Instead of modifying the method just to test it, we could check the test result in another way (**strncmp()** for example).

Reply:
This function never validates whether the plaintext variable is null or not. We should check this at the beginning. If we don't check a null pointer could produce a `SEGV`